virtual_rainforest/main.py

```diff
@@ -13,15 +13,16 @@
 import pint
 from numpy import datetime64, timedelta64
 
-from virtual_rainforest.core.base_model import MODEL_REGISTRY, BaseModel
+from virtual_rainforest.core.base_model import BaseModel
 from virtual_rainforest.core.config import Config
 from virtual_rainforest.core.data import Data, merge_continuous_data_files
 from virtual_rainforest.core.exceptions import ConfigurationError, InitialisationError
 from virtual_rainforest.core.grid import Grid
 from virtual_rainforest.core.logger import LOGGER, add_file_logger, remove_file_logger
+from virtual_rainforest.core.registry import MODULE_REGISTRY
 
 
-def select_models(model_list: list[str]) -> list[Type[BaseModel]]:
+def select_models(model_list: list[str]) -> list[Any]:  # FIXME -> list[type[BaseModel]]
```

Comment:
Why was this ok before and it is not now? What has changed that causes circular import erorrs?

Reply:
The problem is that `registry` is now imported, and if that imports `BaseModel` to type `ModuleInfo.model: BaseModel` then we get a circular import:

```python
 ~/vr_example/ --outpath ~/vr_example/out1 --log ~/vr_example/out1/log.log
Traceback (most recent call last):
  File "/Users/dorme/Library/Caches/pypoetry/virtualenvs/virtual-rainforest-Laomc1u4-py3.10/bin/vr_run", line 3, in <module>
    from virtual_rainforest.entry_points import vr_run_cli
  File "/Users/dorme/Research/Virtual_Rainforest/virtual_rainforest/virtual_rainforest/entry_points.py", line 16, in <module>
    from virtual_rainforest.core.config import config_merge
  File "/Users/dorme/Research/Virtual_Rainforest/virtual_rainforest/virtual_rainforest/core/config.py", line 41, in <module>
    from virtual_rainforest.core.registry import MODULE_REGISTRY, register_module
  File "/Users/dorme/Research/Virtual_Rainforest/virtual_rainforest/virtual_rainforest/core/registry.py", line 27, in <module>
    from virtual_rainforest.core.base_model import BaseModel
  File "/Users/dorme/Research/Virtual_Rainforest/virtual_rainforest/virtual_rainforest/core/base_model.py", line 83, in <module>
    from virtual_rainforest.core.axes import AXIS_VALIDATORS
  File "/Users/dorme/Research/Virtual_Rainforest/virtual_rainforest/virtual_rainforest/core/axes.py", line 53, in <module>
    from virtual_rainforest.core.grid import Grid
  File "/Users/dorme/Research/Virtual_Rainforest/virtual_rainforest/virtual_rainforest/core/grid.py", line 24, in <module>
    from virtual_rainforest.core.config import Config, ConfigurationError
ImportError: cannot import name 'Config' from partially initialized module 'virtual_rainforest.core.config' (most likely due to a circular import) (/Users/dorme/Research/Virtual_Rainforest/virtual_rainforest/virtual_rainforest/core/config.py)
```